### ISSUE TITLE
libcap (2.52 and above) does not compile on RISC-V.

### ISSUE THREAD

**Issue by Madouura** [NixOS/nixpkgs#168813]
### Describe the bug
As the title states, it does not (cross) compile to riscv64.
2.51 is the last working version, I am unsure as to why it will not compile.

```bash
Checking uname version type              : #1-NixOS SMP PREEMPT Fri Apr 8 12:06:58 UTC 2022 
make libcap.so
make[2]: Entering directory '/build/libcap-2.63/libcap'
riscv64-unknown-linux-gnu-gcc -O2  -Wall -Wwrite-strings -Wpointer-arith -Wcast-qual -Wcast-align -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Winline -Wshadow -Wunreachable-code -fPIC -Dlinux -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/build/libcap-2.63/libcap/../libcap/include/uapi -I/build/libcap-2.63/libcap/../libcap/include   empty.c   -o empty
Checking for header stdio.h              : yes 
objcopy --dump-section .interp=loader.txt empty /dev/null
Checking uname version type              : #1-NixOS SMP PREEMPT Fri Apr 8 12:06:58 UTC 2022 
objcopy: Unable to recognise the format of the input file `empty'
make[2]: *** [Makefile:112: loader.txt] Error 1
make[2]: Leaving directory '/build/libcap-2.63/libcap'
make[1]: *** [Makefile:41: all] Error 2
make[1]: Leaving directory '/build/libcap-2.63/libcap'
make: *** [Makefile:12: all] Error 2
builder for '/nix/store/j9g3p9sdwaa2r8xpq93zlcq70k3xi39v-libcap-riscv64-unknown-linux-gnu-2.63.drv' failed with exit code 2
cannot build derivation '/nix/store/h4jldij8c5088qh7nf2p65s3cp1h6npq-nixos-system-nixos-22.05pre-git.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bvw64pr0m0fnx7wbi337ym9n3likrkqm-ext4-fs.img.zst-riscv64-unknown-linux-gnu.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/s19vfi7vyqlaaf337hxvfbybci5l42g5-nixos-sd-image-22.05pre-git-riscv64-linux.img-riscv64-unknown-linux-gnu.drv': 1 dependencies couldn't be built
error: build of '/nix/store/s19vfi7vyqlaaf337hxvfbybci5l42g5-nixos-sd-image-22.05pre-git-riscv64-linux.img-riscv64-unknown-linux-gnu.drv' failed
```

### Steps To Reproduce
1. Set up cross compilation either in nix-build or in your system.
2. Attempt to compile libcap
3. ???
4. It's borked!

### Expected behavior
It should have compiled.

### Additional context
Currently attempting a cross-compile for VisionFive.

### Notify maintainers

@zowoq
### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.19, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11, home-manager-21.11"`
 - channels(mado): `"home-manager-21.11, nixos-21.11.337053.2f06b87f64b, nixos-unstable-22.05pre369117.ff9efb0724d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by tpwrules** [comment]
I imagine this will be fixed by #168303.

**Comment by Madouura** [comment]
Looks like you're right.
